### PR TITLE
Chore: Upgrade react, react-dom, react-test-renderer versions

### PR DESCRIPTION
**Pull request by boyskila**
**What this PR does / why we need it**:
   Upgrade react, react-dom and react-test-renderer versions as part of https://github.com/grafana/grafana/issues/18503

Update: Deprecated ```react-hooks-testing-library``` was replaced with recommended by authors ```@testing-library/react-hooks```

### CODE REVIEW

**Comment by boyskila** [comment]
@davkal @dprokop can you please check, thanks!

**Review by torkelo — APPROVED** [review]
Looks good

File: public/app/plugins/datasource/loki/components/useLokiSyntax.test.ts [hunk 1]
```diff
@@ -66,8 +66,6 @@ describe('useLokiSyntax hook', () => {
 
     act(() => result.current.setActiveOption([activeOptionMock]));
 
-    await waitForNextUpdate();
-
-    expect(result.current.logLabelOptions).toEqual(logLabelOptionsMock3);
+    setTimeout(() => expect(result.current.logLabelOptions).toEqual(logLabelOptionsMock3));
```
Comment (dprokop) [inline public/app/plugins/datasource/loki/components/useLokiSyntax.test.ts]: @boyskila can you elaborate on this change? Can't see how this relates to dependencies update

**Comment by dprokop** [comment]
@torkelo thanks for review. We need to run packages build and e2e before proceeding with this.

File: public/app/plugins/datasource/loki/components/useLokiSyntax.test.ts (hunk 1 above)
Comment (boyskila) [reply public/app/plugins/datasource/loki/components/useLokiSyntax.test.ts]: Just revert it. See my last commit

**Review by hugohaggmark — APPROVED** [review]
LGTM, and e2e tests passed locally!

**Review by peterholmberg — APPROVED** [review]
A few warnings regarding lifecycle methods but no failures on unit tests or building packages.

**Comment by boyskila** [comment]
Thanks, guys for the review! @peterholmberg I guess replacing the UNSAFE methods will solve the warning issues.
So I think this can be merged if you don't have any other comments.

**Comment by dprokop** [comment]
@boyskila thanks again for the contribution!

**Pull request merged**